src/configuration.ts

```diff
@@ -172,6 +173,39 @@ export class ConfigurationManager implements vscode.Disposable
         return {arch: arch, os: os};
     }
 
+    private async getVcpkgPathFromEnv()
+    {
+        // let envVar = process.env.VCPKG_PATH;
+        let envVar = this._context.environmentVariableCollection.get(this._vcpkgRootConfig);
+
+        if (envVar !== undefined && envVar.value !== undefined && envVar.value.length !== 0 && await this.isVcpkgExistInPath(envVar.value))
+        {
+            return envVar.value;
+        }
+
+        return undefined;
+    }
+
+    private async getVcpkgPath()
```

Comment:
We may need to create a new structure `vcpkgPath` and return with it in the future since it should contains one of 2 parts:
- $ENV{VCPKG_ROOT}
- real absoulte path

Reply:
How about this: return the user's write string directly and check in `isVcpkgExistInPath` whether it is an environment variable.
So we don't need to care about such thing in any functions except the validation function.

Reply:
Sounds good.

Reply:
So will that mean if the extension detects env variable, it will ignore the path string user provides in extension settings?

Reply:
> So will that mean if the extension detects env variable, it will ignore the path string user provides in extension settings?

No, first we use a user-supplied string (either an environment variable or an absolute path), otherwise exntesion will check the default environment variable `${VCPKG_ROOT}`.

Reply:
I am a bit confused about this. Because in the extension settings UI, you are not allowed to leave the path field blank. Also you mentioned we cannot use env variables in the path field in extension settings. So will there be a way to know where the path has been taken from. Sorry if I am asking too many questions.

Reply:
Now we support to write environment variables in path field.